arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentTableViewController.swift

```diff
@@ -14,40 +14,74 @@
 
 import UIKit
 
-class ContentTableViewController: UITableViewController, CustomSearchHeaderViewDelegate, DownloadProgressViewDelegate {
+class ContentTableViewController: UITableViewController {
+    
     var samples = [Sample]()
     private var expandedRowIndex:Int = -1
     
-    private var headerView:CustomSearchHeaderView!
     var containsSearchResults = false
     private var bundleResourceRequest:NSBundleResourceRequest!
     private var downloadProgressView:DownloadProgressView!
-
-    var token: Int = 0
+    
+    var searchEngine: SearchEngine?
+    
+    // strong reference needed for iOS 10
+    var filterSearchController:UISearchController?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        self.tableView.rowHeight = UITableView.automaticDimension
-        self.tableView.estimatedRowHeight = 60
-        
-        if containsSearchResults {
-            self.tableView.tableHeaderView?.removeFromSuperview()
-            self.tableView.tableHeaderView = nil
-        }
-        else {
-            self.headerView = self.tableView.tableHeaderView! as? CustomSearchHeaderView
-            self.headerView.delegate = self
-            self.headerView.hideSuggestionsTable()
-        }
+        searchEngine = SearchEngine(samples: samples)
         
         //initialize download progress view
-        self.downloadProgressView = DownloadProgressView()
-        self.downloadProgressView.delegate = self
+        downloadProgressView = DownloadProgressView()
+        downloadProgressView.delegate = self
+        
+        if !containsSearchResults{
+            addFilterSearchController()
+        }
     }
     
-    func samplesByNames<C: Collection>(_ names: C) -> [Sample] where C.Element == String {
-        return samples.filter { names.contains($0.name) }
+    private func addFilterSearchController(){
+        
+        // ensure that the search results are interactable
+        definesPresentationContext = true
+        
+        // create the search controller
+        let searchController = UISearchController(searchResultsController:nil)
+        searchController.obscuresBackgroundDuringPresentation = false
+        searchController.hidesNavigationBarDuringPresentation = false
+        // send search query updates to the results controller
+        searchController.searchResultsUpdater = self
+        // retain a strong reference for iOS 10
+        self.filterSearchController = searchController
+        
+        let searchBar = searchController.searchBar
+        searchBar.placeholder = "Filter"
```

Comment:
I think Search would still be an appropriate placeholder. You are just searching through a subset of all samples, namely the samples in the selected category.

Reply:
Despite both fields searching, I used "Filter" to indicate that this search bar is functionally different than the one on the collection view. Otherwise they look identical. I didn't know there was a difference when I first used the app (with the old search bars that were identical).

Reply:
Actually, I prefer the Filter text within a category since it describes what's happening. It might be more intuitive to replace the magnifying glass with a filter icon to match the placeholder text, something like this (but obviously with a better icon):

![2018-09-14_15-58-07](https://user-images.githubusercontent.com/25374/45572685-ffda2a80-b837-11e8-93b4-a78ffcfe9a7c.png)

Reply:
Or, as with the HIG Stock Search sample, provide some text to indicate you're only looking for samples in the current category:

![2018-09-14_16-14-09](https://user-images.githubusercontent.com/25374/45573037-382e3880-b839-11e8-9c16-d8fb79ac14e2.png)